### PR TITLE
[quant][graphmode][fx] Embedding/EmbeddingBag works in static quant environment

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47068 [quant][graphmode][fx][fix] Linear work with float_qparam_dynamic_qconfig
* **#47029 [quant][graphmode][fx] Embedding/EmbeddingBag works in static quant environment**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24607958](https://our.internmc.facebook.com/intern/diff/D24607958)

### CODE REVIEW

File: torch/quantization/fx/pattern_utils.py [hunk 1]
```diff
@@ -37,6 +37,18 @@ def get_default_quant_patterns():
 def get_default_output_activation_post_process_map():
     return DEFAULT_OUTPUT_ACTIVATION_POST_PROCESS_MAP
 
+# a set of QuantizeHandler classes that are not observed
+# we'll skip inserting observers for input and output for these QuantizeHandlers
+# used for ops that only supports dynamic/weight only quantization
```
Comment (supriyar) [inline torch/quantization/fx/pattern_utils.py:42]: We seem to be only  using this for weight-only quant right now. Do we plan on using this for dynamic quant as well?

Comment (jerryzh168) [reply torch/quantization/fx/pattern_utils.py:42]: yeah this should be used for ops that **only** support dynamic quantization as well

File: test/quantization/test_quantize_fx.py [hunk 2]
```diff
@@ -1695,15 +1695,23 @@ def forward(self, indices):
...
-            model,
-            [[indices]],
-            QuantType.DYNAMIC,
-            quantized_node,
-            custom_qconfig=float_qparams_dynamic_qconfig
-        )
-
-    def test_qembedding_bag_module(self):
+        configs = [
+            (float_qparams_dynamic_qconfig, ns.call_module(nnq.Embedding)),
```
Comment (supriyar) [inline test/quantization/test_quantize_fx.py:1699]: What would happen in the case where the model has say a Linear layer and Embedding layer and we set the qconfig to `float_qparams_dynamic_qconfig `? We need to make sure in that case Linear doesn't get quantized as well.

Comment (jerryzh168) [reply test/quantization/test_quantize_fx.py:1699]: not sure if we are missing some checks, but this can be done in a separate pr

Comment (supriyar) [reply test/quantization/test_quantize_fx.py:1699]: Sounds good. Should we extend this test to include another static quant op in it (say linear) and test with different qconfig to make sure the right layers are quantized?

**Review by supriyar — APPROVED** [review]
Looks good, left a comment about the testing.

File: test/quantization/test_quantize_fx.py (hunk 2 above)
Comment (jerryzh168) [reply test/quantization/test_quantize_fx.py:1699]: you mean nn.linear with float_qparams_dynamic_qconfig right? will test this separately in a next pr.

**Comment by jerryzh168** [comment]
User is responsible to make sure the qconfig is correct for embeddings. closing in favor of: https://github.com/pytorch/pytorch/pull/47092

**Pull request closed without merging**